pkg/workspace/library_module.go

```diff
@@ -237,7 +246,23 @@ func (l *libraryValue) Eval(thread *starlark.Thread, f *starlark.Builtin,
 		return starlark.None, fmt.Errorf("expected no arguments")
 	}
 
-	libraryExecution := l.libraryExecutionFactory.New(l.libraryCtx)
+	allowedKWArgs := map[string]struct{}{
+		"data_values_disable_validation": {},
+	}
+	if err := core.CheckArgNames(kwargs, allowedKWArgs); err != nil {
+		return starlark.None, err
+	}
+
+	skipDataValuesValidation, err := core.BoolArg(kwargs, "data_values_disable_validation")
+	if err != nil {
+		return starlark.None, err
+	}
+
+	libraryExecutionFactory := l.libraryExecutionFactory
+	if skipDataValuesValidation {
+		libraryExecutionFactory = l.libraryExecutionFactory.ThatSkipsDataValuesValidations()
+	}
+	libraryExecution := libraryExecutionFactory.New(l.libraryCtx)
```

Comment:
Very good catch. You're absolutely right. This flag should live on `@ytt:library.get()` not `@ytt:library.eval()`; otherwise, the author could receive data or results that are invalid.

🙌🏻